File: src/main/scala/it/unibo/bd18/stacklite/QuestionTagData.scala
package it.unibo.bd18.stacklite

sealed trait QuestionTagData {

  def id: Int

  def tag: String

  def toCSVString: String = s"$id,$tag"

}

object QuestionTagData {

  def apply(row: Array[String]): QuestionTagData = QuestionTagDataImpl(
    getId(row),
    getTag(row))

  private def getId(row: Array[String]): Int = row(0).toInt

  private def getTag(row: Array[String]): String = row(1)

  private case class QuestionTagDataImpl(
                                          override val id: Int,
                                          override val tag: String
                                        ) extends QuestionTagData

}

File: src/main/scala/it/unibo/bd18/stacklite/QuestionTagData.java
package it.unibo.bd18.stacklite;

import java.io.Serializable;

public final class QuestionTagData implements Serializable {

    private final int id;
    private final String tag;

    public static QuestionTagData create(String row) {
        return create(row.split("\\s*,\\s*"));
    }

    public static QuestionTagData create(String[] row) {
        return new QuestionTagData(
                Utils.readInt(row[0]),
                row[1]);
    }

    public QuestionTagData(int id, String tag) {
        this.id = id;
        this.tag = tag;
    }

    public int id() {
        return id;
    }

    public String tag() {
        return tag;
    }

    public String toCSVString() {
        return String.format("%d,%s", id, tag);
    }

}
